### PR TITLE
cleanups

### LINKED ISSUE #19: Implement bitfield data_structure for attester_bitfield
Currently we are using an IntSet for the attester_bitfield field. However this is not ideal.

### Cons

- The SSZ serialization format of the bitfield is the raw packed bytes
- [IntSet](https://nim-lang.org/docs/intsets.html) uses an [array of 34 ints on the stack](https://github.com/nim-lang/Nim/blob/72e15ff739cc73fbf6e3090756d3f9cb3d5af2fa/lib/pure/collections/intsets.nim#L139) and the rest on the heap. The first 34 ints are not bit-packed, the tail on the heap is packed. So we're always in the worst case scenario as the [minimum recommended committee size is 111](https://vitalik.ca/files/Ithaca201807_Sharding.pdf) and the [TARGET_COMMITTEE_SIZE is 256](https://github.com/ethereum/eth2.0-specs/blob/f1b78cf01e6250bd2b206e2b173111a5e4f063b9/specs/core/0_beacon-chain.md)

### Issues with Nim builtin sets

- Taking the length of it requires iterating on all bits
- It can hold at most 2^16 ints, the maximum committee size according to specs must be checked
- Serialization is probably a binary dump but that must be tested

### Pros 

- It's battle-tested and fast.
- The length can be solved by using a custom object that tracks the length
```Nim
type MyBitfield = object
  data: set[int16]
  len: int16
```

## Do-It-Yourself bitfield

We can be sure that it works for our purposes but extra implementation, testing and maintenance time.

Python implementation is [a short 36 lines here](https://github.com/ethereum/beacon_chain/blob/fd86633467f0cedaba838201d98a2ddcfdedc7db/beacon_chain/utils/bitfield.py). [Tests](https://github.com/ethereum/beacon_chain/blob/master/tests/utils/test_bitfield.py).

Note that the beacon chain specs mentions `len(attester_bitfield) == ceil_div8(len(attestation_indices))` with `ceil_div8 = (x + 7) // 8`, This checks that the byte length of the bitfield is equal to the number of attesters divided by 8 and rounded up to the nearest multiple of 8. An alternative and arguably clearer implementation can be derived from Nimbus [ceil32](https://github.com/status-im/nimbus/blob/f7d858d58bc095dab9ca26d7577b3d8a7cea6efc/nimbus/vm/interpreter/utils/utils_numeric.nim#L22)

### PR DESCRIPTION
**Pull request by arnetheduck**
* add simple bitfield type (fixes #19)
* fix bit endianess to match spec
* consolidate attestation and block logging
* cruft cleanup
* run optimizer on tests, speeds up build

### CODE REVIEW

**Comment by zah** [comment]
So, why didn't you use the time to fix the naming issues in nim-ranges instead? The code there already supports both addressing modes, so most of the new code here could have been avoided.

File: beacon_chain/spec/bitfield.nim [hunk 1]
```diff
@@ -0,0 +1,45 @@
+type
+  BitField* = object
```
Comment (zah) [inline beacon_chain/spec/bitfield.nim]: You haven't defined the Json serialization for this type. The default one will be a bit unusual - as an array of numbers.

**Comment by arnetheduck** [comment]
because I want to insulate the spec code from external changes, similar to `crypto.nim` and `digest.nim` and retain the spec naming for functions. mostly the same reason as before - without test cases of the spec, we can only rely on ocular inspection - this code merely moves it all in one place so it's easy to refactor later.

File: beacon_chain/spec/bitfield.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/spec/bitfield.nim]: I'm waiting for the upstream tests to define one: https://github.com/ethereum/eth2.0-tests/tree/master/state

**Pull request merged**